### ISSUE TITLE
Linux ARM64 (Native bindings for Linux ARM64)

### ISSUE THREAD

**Issue by DJSures** [shimat/opencvsharp#1644]
# Summary of your issue

OpenCVSharp is fantastic; kudos to all your hard work! My cross-platform experience is unfortunately limited because there are no Linux ARM64 native bindings. I'm surprised to see 32-bit ARM NuGet packages (OpenCvSharp4.runtime. Linux-arm) in 2024. Many of the OpenCVSharp user issues in this repo are because of the lack of ARM64 bindings. All Raspberry Pi and ARM SBC Linux's have favored 64bit for the last few years. You will be hard-pressed to find a 32-bit OS these days.

# Environment

- Raspberry Pi Zero 2 (BCM2710) is 64 bit
- Raspberry Pi 3 (BCM2710) is 64 bit
- Raspberry Pi 4 (BCM2711) is 64 bit
- And, of course, Raspberry Pi 5 is 64-bit

# What did you do when you faced the problem?

I started Sherlock'n the interwebs to find a libOpenCvSharpExtern.so: ELF 64-bit LSB shared object, ARM aarch64 compatible with OpenCVSharp4 (not SDCB's Mini, because /dev/videoX is essential)

Until there's a build for Linux ARM64, I'll have to build it myself - but I'd prefer a shared Nuget that solves the many user issues in this repo.

## Example code:

Build it yourself RN: https://docs.opencv.org/4.x/d7/d9f/tutorial_linux_install.html

## Output:

*sad face emoji*

## What did you intend to be?

Like all children, an astronaut, cowboy, firefighter, or robot. I made a few million robots possible, so I kind of achieved my goal! What did you intend to be?

**Comment by sajmons** [comment]
Thumbs up for ARM64 NuGet package for raspberry Pi!

**Comment by DJSures** [comment]
Did you build it yourself or do you want me to post the one I built?

**Comment by sajmons** [comment]
No, I didn't build it myself, I yust wanna see it build as NuGet package so that can be easily included for linux-arm64 Raspberry Pi projects.

**Comment by andywcoder** [comment]
> Did you build it yourself or do you want me to post the one I built?

Can you please post the one you built?

**Comment by DJSures** [comment]
Here you go
[libOpenCvSharpExtern.zip](https://github.com/shimat/opencvsharp/files/14861749/libOpenCvSharpExtern.zip)

**Comment by andywcoder** [comment]
Thanks for the library. Unfortunately it is dynamic linked unlike the static linked library in the nuget package OpenCvSharp4.runtime.linux-arm.

I built the OpenCV 4.9.0 binaries which took half a day on my Raspberry Pi 4. After that `ldd /libOpenCvSharpExtern.so` still displays a missing reference:

`/lib/aarch64-linux-gnu/libstdc++.so.6: version 'GLIBCXX_3.4.32' not found`

Running `strings /lib/aarch64-linux-gnu/libstdc++.so.6 | grep GLIBCXX` prints

```
...
GLIBCXX_3.4.27
GLIBCXX_3.4.28
GLIBCXX_3.4.29
GLIBCXX_3.4.30
```

So I'm trying to update libstdc++.so.6 which apparently is part of GCC. After building the GCC 13.2 binaries for a couple of hours it stopped with an error about an `symbol lookup error: /lib/aarch64-linux-gnu/libicuuc.so.72: undefined symbol: _ZSt15__once_callable, version GLIBCXX_3.4.11`.

During all the building I rewrote the C# app in Python, which is much easier to run on Linux. I really like C# and .NET, but things like that are just a hassle I don't need.

**Comment by mfitconsultants** [comment]
Hi, 

Does anyone know if there are any updates or instructions on this? I am trying to implement this on a Raspberry Pi 5 for arm64.

I have ran the scripts here

[https://docs.opencv.org/4.x/d7/d9f/tutorial_linux_install.html#tutorial_linux_install_detailed_basic_verify](https://docs.opencv.org/4.x/d7/d9f/tutorial_linux_install.html#tutorial_linux_install_detailed_basic_verify) 

And compiled all the binaries. however I cannot see `libOpenCvSharpExtern.so` or `OpenCvSharpExtern.so` present anywhere?

Therefore I tried downloading the attached `libOpenCvSharpExtern.so` dropping it in the `build/lib` directory and running `lld libOpenCvSharpExtern.so`  where I just get;

> ./libOpenCvSharpExtern.so: /lib/aarch64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.32' not found (required by ./libOpenCvSharpExtern.so)
>         linux-vdso.so.1 (0x00007fff06820000)
>         libopencv_stitching.so.409 => not found
>         libopencv_aruco.so.409 => not found
>         libopencv_bgsegm.so.409 => not found
>         libopencv_dnn_superres.so.409 => not found
>         libopencv_face.so.409 => not found
>         libopencv_img_hash.so.409 => not found
>         libopencv_line_descriptor.so.409 => not found
>         libopencv_quality.so.409 => not found
>         libopencv_superres.so.409 => not found
>         libopencv_tracking.so.409 => not found
>         libopencv_wechat_qrcode.so.409 => not found
>         libopencv_xfeatures2d.so.409 => not found
>         libopencv_xphoto.so.409 => not found
>         libopencv_shape.so.409 => not found
>         libopencv_highgui.so.409 => not found
>         libopencv_text.so.409 => not found
>         libopencv_ml.so.409 => not found
>         libopencv_optflow.so.409 => not found
>         libopencv_ximgproc.so.409 => not found
>         libopencv_video.so.409 => not found
>         libopencv_videoio.so.409 => not found
>         libopencv_imgcodecs.so.409 => not found
>         libopencv_objdetect.so.409 => not found
>         libopencv_calib3d.so.409 => not found
>         libopencv_dnn.so.409 => not found
>         libopencv_features2d.so.409 => not found
>         libopencv_flann.so.409 => not found
>         libopencv_photo.so.409 => not found
>         libopencv_imgproc.so.409 => not found
>         libopencv_core.so.409 => not found
>         libstdc++.so.6 => /lib/aarch64-linux-gnu/libstdc++.so.6 (0x00007fff06270000)
>         libm.so.6 => /lib/aarch64-linux-gnu/libm.so.6 (0x00007fff061d0000)
>         libgcc_s.so.1 => /lib/aarch64-linux-gnu/libgcc_s.so.1 (0x00007fff06190000)
>         libc.so.6 => /lib/aarch64-linux-gnu/libc.so.6 (0x00007fff05fd0000)
>         /lib/ld-linux-aarch64.so.1 (0x00007fff067e8000)

Can anyonw direct me as to where I am going wrong at all?
